src/sdk/namespace_fs.js

```diff
@@ -402,7 +421,11 @@ class NamespaceFS {
                         return; // not added
                     }
 
-                    if (!delimiter && r.common_prefix) {
+                    if (ent.name === HIDDEN_VERSIONS_PATH && r.common_prefix) {
+                        if (list_versions) {
+                            await this._list_version_dir_objects(fs_context, versions, ent.name, dir_key);
```

Comment:
as we discussed before, can't we just use process_dir ?
in process_dir() we use the dir_cache which makes the performance much better and provides sorting as well, the readdir exists also there in the cache

Reply:
Yes, this is taken care.